includes/optimizer.php

```diff
@@ -0,0 +1,162 @@
+			/**
+			 * If a dev has added allowed tags or attributes, we should not optimize the SVGs, because the optimizer will not respect their exclusions.
+			 */
+			if ( $has_svg_allowed_tags || $has_svg_allowed_attributes ) {
+				return false;
+			}
+			$params = $this->svgo_params();
+			return ( ! empty( $params ) && is_array( $params ) );
+		}
+		/**
+		 * The SVGO parameters. Developers can use this filter to pass additional parameters or completely disable the optimizer by passing:
+		 * add_filter( 'safe_svg_svgo_params', '__return_false' );
+		 *
+		 * @return mixed|null
+		 */
+		public function svgo_params() {
+			return apply_filters(
+				'safe_svg_svgo_params',
+				[
+					'multipass' => true,
+				]
+			);
+		}
+		/**
+		 * Enqueue the necessary scripts.
+		 *
+		 * @param string $hook The current admin page.
+		 *
+		 * @return void
+		 */
+		public function enqueues( $hook ) {
+			$allowed_hooks = [
+				'options-media.php',
+				'post.php',
+				'post-new.php',
+				'upload.php',
+				'media-new.php',
+			];
+			if ( ! in_array( $hook, $allowed_hooks, true ) ) {
+				return;
+			}
+			wp_enqueue_script(
+				'safe-svg-admin-scripts',
+				SAFE_SVG_PLUGIN_URL . '/dist/safe-svg-admin.js',
+				[ 'wp-data' ],
+				SAFE_SVG_VERSION,
+				true
+			);
+			$params = wp_json_encode(
+				[
+					'ajaxUrl'    => esc_url( admin_url( 'admin-ajax.php' ) ),
+					'svgoParams' => wp_json_encode( $this->svgo_params() ),
+					'nonce'      => wp_create_nonce( $this->nonce_name ),
+					'context'    => $hook,
+				]
+			);
+			wp_add_inline_script(
+				'safe-svg-admin-scripts',
+				sprintf(
+					'var safeSvgParams = %s;',
+					$params
+				),
+				'before'
+			);
+		}
+		/**
+		 * Optimize the SVG file.
+		 *
+		 * @return void
+		 */
+		public function optimize() {
+			$svg_url = filter_input( INPUT_GET, 'svg_url', FILTER_SANITIZE_URL );
+			if ( ! current_user_can( 'edit_posts', attachment_url_to_postid( $svg_url ) ) ) {
+				return;
+			}
+			check_ajax_referer( $this->nonce_name, 'svg_nonce' );
+			$svg_path = $this->url_to_path( $svg_url );
+			if ( empty( $svg_path ) ) {
+				return;
+			}
+			$maybe_dirty = $_GET['optimized_svg'];
+			$sanitizer   = new Sanitizer();
+			$sanitizer->minify( true );
+			$sanitized = $sanitizer->sanitize( stripcslashes( $maybe_dirty ) );
+
+			if ( empty( $sanitized ) ) {
+				return;
+			}
+			file_put_contents( $svg_path, $sanitized ); // phpcs:ignore WordPress.WP.AlternativeFunctions.file_system_read_file_put_contents
+			wp_die();
+		}
+		/**
+		 * A helper method to get the file path from its URL.
+		 *
+		 * @param string $url The URL string.
+		 *
+		 * @return false|string
+		 */
+		protected function url_to_path( string $url = '' ) {
+			if ( empty( $url ) ) {
+				return '';
+			}
+			$parsed_url = wp_parse_url( $url );
+			if ( empty( $parsed_url['path'] ) ) {
+				return false;
+			}
+			$file = ABSPATH . ltrim( $parsed_url['path'], '/' );
```

Comment:
I figured out why I wasn't getting the optimised version of the image.

I was testing on a site running WP in [sub-directory mode](https://wordpress.org/documentation/article/giving-wordpress-its-own-directory/) with the directory structure:
```
/ #root directory
/content #content directory
/wp # WordPress install.
```

Instead of `path/to/content/.../file.svg` the `$file` is calculated as `path/to/wp/content/.../file.svg`.

In the method above, I've suggested you use `get_attached_file()` which will remove the need for this method.

Reply:
Thanks for that @peterwilsoncc I followed your suggestion and on my local tests it seems to be working.